homeassistant/components/viam/strings.json

```diff
@@ -0,0 +1,209 @@
+{
+  "config": {
+    "step": {
+      "user": {
+        "title": "Authenticate with Viam",
+        "description": "Select which credential type to use.",
+        "data": {
+          "credential_type": "Credential Type"
+        }
+      },
+      "auth": {
+        "title": "[%key:component::viam::config::step::user::title%]",
+        "description": "Provide the credentials for communicating with the Viam service.",
+        "data": {
+          "api_id": "API Key ID",
+          "api_key": "API Key Value",
+          "address": "Robot Address",
+          "secret": "Robot Secret"
+        },
+        "data_description": {
+          "address": "Find this under the Code Sample tab in the app.",
+          "secret": "Find this under the Code Sample tab in the app when 'include secret' is enabled."
+        }
+      },
+      "robot": {
+        "data": {
+          "robot": "Select a robot"
+        }
+      }
+    },
+    "error": {
+      "cannot_connect": "[%key:common::config_flow::error::cannot_connect%]",
+      "invalid_auth": "[%key:common::config_flow::error::invalid_auth%]",
+      "unknown": "[%key:common::config_flow::error::unknown%]"
+    }
+  },
+  "selector": {
+    "credential_type": {
+      "options": {
+        "api-key": "Org API Key",
+        "robot-location-secret": "Robot Location Secret"
+      }
+    }
+  },
+  "exceptions": {
+    "entry_not_found": {
+      "message": "No Viam config entries found"
+    },
+    "entry_not_loaded": {
+      "message": "{config_entry_title} is not loaded"
+    },
+    "invalid_config_entry": {
+      "message": "Invalid config entry provided. Got {config_entry}"
+    },
+    "unloaded_config_entry": {
+      "message": "Invalid config entry provided. {config_entry} is not loaded."
+    },
+    "robot_credentials_required": {
+      "message": "The robot address is required for this connection type."
+    },
+    "robot_credentials_not_found": {
+      "message": "The necessary credentials for the RobotClient could not be found."
+    }
+  },
+  "services": {
+    "common": {
+      "name": "Common fields for Viam integration services",
+      "description": "Only used for string translations",
```

Comment:
We don't usually do like this, I'd like a second opinion on if this is OK

Reply:
I believe it helps prevent typos or forgotten changes across these repeated values, similar to the global `key:common::config_flow` references across other `strings.json` within integrations.

Reply:
Yes, it's fine - and encouraged - to reference withing the `strings.json`-file, but we don't usually add a fake section.

Reply:
I did have a discussion with some core developers. We think it's a great idea to add deduplication support, but we don't think it's OK to add a fake service for that.
Instead, we'd like to see a new top level dictionary where common strings can be added.

Would you like to pick that up? You'd need to update `script/hassfest/translations.py` with support for the new `common` dictionary. Hassfest should also make sure the common dictionary does not have orphan strings.

This should be done in a separate PR. You can remove the use of the fake service from this PR for now if you don't want that improvement to bock this PR.

Reply:
Removed the `common` service dictionary. I'll take a look at updating the script to allow for a top-level `common` in `strings.json` in a separate PR. Thanks for reviewing!